File: back-end/src/main/java/server/model/Korisnik.java
package server.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Korisnik {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String korisnickoIme;

    @Column(nullable = false)
    private String lozinka;

    @Column(nullable = false, unique = true)
    private String email;
	
}

File: front-end/src/app/models/Korisnik.ts
import { DodeljenoPravoPristupa } from "./DodeljenoPravoPristupa";

export interface Korisnik {
  id: number;
  email: string;
  lozinka: string;
  dodeljenaPravaPristupa?: DodeljenoPravoPristupa[];
  vidljiv: boolean;
}
